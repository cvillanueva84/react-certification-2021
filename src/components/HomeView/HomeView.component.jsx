import React from 'react';
import styled from 'styled-components';

import VideoCard from '../VideoCard';

import './HomeView.styles.css';

import youtubeJson from '../../utils/mock/youtube-videos-mock.json';

const HomeViewContainer = styled.div`
  background-color: white;
  display: flex;
  flex-wrap: wrap;
  font-size: 18px;
  height: calc(100vh - 3.5em);
  justify-content: center;
  overflow-x: hidden;
`;

function HomeView() {
  const youtubeVideos = 'items' in youtubeJson ? youtubeJson.items : [];

  return (
    <HomeViewContainer className="home-view-container">
      {youtubeVideos
        .filter((videoInfo) => videoInfo.id.kind.includes('video'))
        .map((videoInfo) => {
          return (
            <VideoCard
              title={videoInfo.snippet.title}
              thumbnail={videoInfo.snippet.thumbnails.medium.url}
              key={videoInfo.etag}
            />
          );
        })}
    </HomeViewContainer>
  );
}

export default HomeView;
