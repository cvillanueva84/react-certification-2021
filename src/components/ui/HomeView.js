import React from 'react'
import { HeaderNav } from '../HeaderNav';
import { HomeScreen } from './HomeScreen';

export const HomeView = () => {
    return (
        <>
            <HeaderNav/>
            <HomeScreen/>
        </>
    )
}
