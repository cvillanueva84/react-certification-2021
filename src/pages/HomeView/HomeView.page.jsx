import React from 'react';
import NavBar from '../../components/NavBar';
import VideosContainer from '../../components/VideosContainer';
import { data } from '../../data/youtube-videos-mock';

const HomeView = () => {
  return (
    <>
      <NavBar />
      <h2 className="content-title">Welcome to the Challenge!</h2>
      <VideosContainer {...data} />
    </>
  );
};

export default HomeView;
