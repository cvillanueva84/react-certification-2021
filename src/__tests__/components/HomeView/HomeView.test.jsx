import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import HomeView from '../../../components/HomeView/HomeView.component';

describe('HomeView', () => {
  it('Renders HomeView', () => {
    const tree = renderer.create(<HomeView youtubeData={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
