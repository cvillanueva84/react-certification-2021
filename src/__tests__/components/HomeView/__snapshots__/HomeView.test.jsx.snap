// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView Renders HomeView 1`] = `
<div
  className="sc-bdnxRM gdgpzt home-view-container"
/>
`;
